packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -993,6 +993,99 @@ class DefaultTextStyleTransition extends AnimatedWidget {
 /// ** See code in examples/api/lib/widgets/transitions/animated_builder.0.dart **
 /// {@end-tool}
 ///
+/// {@template flutter.flutter.animatedbuilder_changenotifier.rebuild}
+/// ## Granular rebuilds with AnimatedBuilder and ChangeNotifier
+///
+/// A [State.setState] method call requires Flutter to entirely rebuild the
+/// current widget plus a variable portion of the subtree, depending on the
+/// depth and the structure.
+///
+/// To avoid unnecessary work, the [AnimatedBuilder] class can be used along
+/// with a [ChangeNotifier] to only rebuild certain portions a widget,
+/// leaving other parts and the subtree untouched.
+///
+/// Since a [ChangeNotifier] is a subtype of [Listenable], it can be used in
+/// an [AnimatedBuilder] to listen for changes. For example, to keep track
+/// of the state of a simple counter, a [ChangeNotifier] can be created and
+/// exposed to the widgets below using an [InheritedWidget].
+///
+/// {@tool snippet}
+///
+/// This is an example that shows how an [InheritedWidget] can expose a
+/// [ChangeNotifier] to a widget and its children.
```

Comment:
Why show an example of that in the documentation of AnimatedBuilder?

Reply:
Let's focus on what we actually want to show: How to use AnimatedBuilder with a ChangeNotifier.